File: Agregacion/Ejercicio-9/CarritoCompras.java
import java.util.*;

public class CarritoCompras {
    private List<Producto> productos;
    
    public CarritoCompras()
    {
        this.productos=new ArrayList<>();
    }

    public List<Producto> getProductos() {
        return productos;
    }

    public void setProductos(List<Producto> productos) {
        this.productos = productos;
    }
     
    public void agregarProducto(Producto p)
    {
        productos.add(p);
    }
    
    public void mostrarCarrito()
    {
        System.out.println("----------------------------CARRITO-------------------");
        for(Producto p: productos)
        {
            p.mostrarInfo();
        }
    }
}

class Producto {
    private String nombre;
    private double precio;
    
    public Producto(String n, double p)
    {
        this.nombre=n;
        this.precio=p;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }
    
    public void mostrarInfo()
    {
        System.out.println("Nombre: "+nombre+" "+"Precio: "+precio);
    }
}

class App {
    public static void main(String[] args) {
        
        Producto p1=new Producto("Sandia", 15.7);        
        Producto p2=new Producto("Cereal", 24.3);
        Producto p3=new Producto("Pollo", 30.2);
        Producto p4=new Producto("Oregano", 5.6);
        
        CarritoCompras c1=new CarritoCompras();
        
        c1.agregarProducto(p1);        
        c1.agregarProducto(p2);
        c1.agregarProducto(p3);
        c1.agregarProducto(p4);
        
        c1.mostrarCarrito();
    }
}

File: Agregacion/Ejercicio-9/CarritoCompras.py
class Producto:
    def __init__(self, nombre, precio):
        self.nombre = nombre
        self.precio = precio

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_precio(self):
        return self.precio

    def set_precio(self, precio):
        self.precio = precio

    def mostrar_info(self):
        print(f"Nombre: {self.nombre} Precio: {self.precio}")


class CarritoCompras:
    def __init__(self):
        self.productos = []

    def get_productos(self):
        return self.productos

    def set_productos(self, productos):
        self.productos = productos

    def agregar_producto(self, producto):
        self.productos.append(producto)

    def mostrar_carrito(self):
        print("----------------------------CARRITO-------------------")
        for producto in self.productos:
            producto.mostrar_info()


# MAIN
if __name__ == "__main__":
    p1 = Producto("Sandía", 15.7)
    p2 = Producto("Cereal", 24.3)
    p3 = Producto("Pollo", 30.2)
    p4 = Producto("Orégano", 5.6)

    c1 = CarritoCompras()

    c1.agregar_producto(p1)
    c1.agregar_producto(p2)
    c1.agregar_producto(p3)
    c1.agregar_producto(p4)

    c1.mostrar_carrito()